src/FakeClient/FakeClientMain.cpp

```diff
@@ -57,6 +59,33 @@ void InstallSigintHandler() {
   sigaction(SIGINT, &act, nullptr);
 }
 
+// On OrbitService's side, and in particular in LinuxTracing, the only information needed to
+// instrument a function is what uprobes need, i.e., the path of the module and the function's
+// offset in the module (address minus load bias). But CaptureClient requires much more than that,
+// through the ModuleManager and the FunctionInfos. For now just keep it simple and for the fields
+// that are not needed just use default or fake values.
+void ManipulateModuleManagerAndSelectedFunctionsToAddUprobe(
+    orbit_client_data::ModuleManager* module_manager,
+    absl::flat_hash_map<uint64_t, orbit_client_protos::FunctionInfo>* selected_functions,
+    const std::string& file_path, uint64_t file_offset) {
+  orbit_grpc_protos::ModuleInfo module_info;
+  module_info.set_file_path(file_path);
+  constexpr const char* kFakeBuildId = "id";
+  module_info.set_build_id(kFakeBuildId);
+  constexpr uint64_t kFakeLoadBias = 0;
+  module_info.set_load_bias(kFakeLoadBias);
+  CHECK(module_manager->AddOrUpdateModules({module_info}).empty());
+
+  orbit_client_protos::FunctionInfo function_info;
+  function_info.set_module_path(file_path);
+  function_info.set_module_build_id(kFakeBuildId);
```

Comment:
will this break once we have build_id validation on the service side?

Reply:
Are we planning on adding validation of `InstrumentedFunction.file_build_id` to the service? If yes, then yes. At that point we can add another command line flag for it, or read it from the binary. I'm not particularly worried, but I also don't know what the specific plan is.

Reply:
yes - there is bug for it: http://b/183380723

Reply:
Fine, let's make this cleaner then. It should be easy to get the build id with `ElfFile`, right?

Reply:
yes - it is pretty straightforward, you can also do load_bias correctly here using that.

Reply:
Done.